src/intervals/exact_numbers.jl

```diff
@@ -0,0 +1,68 @@
+struct ExactNumber{T} <: Real
```

Comment:
We would want `Complex{<:ExactNumber}` to be supported, no?
In which case it may be better to have `ExactReal`, or something of the sort.